src/CommunityToolkit/Xamarin.CommunityToolkit/Behaviors/EventToCommandBehaviorGeneric.shared.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+
+namespace Xamarin.CommunityToolkit.Behaviors
+{
+	/// <summary>
+	/// This <see cref="EventToCommandBehavior"/> cast the sender object to a specific type defined by the user. 
+	/// </summary>
+	/// <typeparam name="TType">The type that you want to receive in your <see cref="Xamarin.Forms.Command"/> </typeparam>
+	public sealed class EventToCommandBehavior<TType> : EventToCommandBehavior
+	{
+		protected override void OnTriggerHandled(object sender = null, object eventArgs = null)
+		{
+			var parameter = CommandParameter
+				?? EventArgsConverter?.Convert(eventArgs, typeof(object), null, null)
+				?? eventArgs;
+
+			if (parameter?.GetType() != typeof(TType))
```

Comment:
parameter = 2 | **int**
TType is **int?** (nullable int)

I presume it will return false

Reply:
Good catch, let me try. But the TType will the whatever the user provides...

`EventToCommandBehavior<int> or EventToCommandBehavior<int?>`

Let me do the test and fix it if needed